vizro-core/src/vizro/tables/_dash_ag_grid.py

```diff
@@ -57,6 +57,9 @@ def dash_ag_grid(data_frame: pd.DataFrame, **kwargs) -> dag.AgGrid:
         >>> vm.Page(title="Page", components=[vm.AgGrid(figure=dash_ag_grid(...))])
 
     """
+    if "columnDefs" in kwargs:
```

Comment:
@petar-qb This is a massive speedup as well - if you agree It hink we should implement it, and also for the Dash datatable. Together with your improvements, this will speed up most table implementations

Reply:
@maxschulz-COL Thanks for the optimisation! For the "Page_3" in the example, the server response time is decreased from `9.5s` -> `2.5s`. 

However, after spending some time investigating the `columnDefs` for AgGrid I found a few more ways to define its elements:
1.     {"valueGetter": {"function": "params.data.sepal_length"}, "headerName": "Col_4"}
2.     {
        'headerName': 'Petal',
        'children': [
            {'field': "petal_width", 'headerName': 'Width'},
            {'field': "petal_length", 'headerName': 'Length'},
        ]
    },
3. And like this -> https://dash.plotly.com/dash-ag-grid/column-definitions#accessing-nested-data-values  
4.     {"field": "petal_length", "headerName": "Col_1"},
       {"field": "petal_length", "headerName": "Col_2"},

To accommodate all these combinations, we'd need to implement a complex if-else structure combined with recursion (for nested fields) to handle the search mechanism (searching for the final set of columns for the `rowData`).

Even then, we don't have a perfect solutions because there are some use-cases when the `rowData` has to contain an entire data_frame (no matter what is in the `columnDefs`). For example, if users want to implement a `custom filter interaction`, where the target is filtered by a `hidden column` when the source agGrid row data is clicked. With this optimisation, this `hidden column` would be preemptively deleted and callback would raise an exception. 

So, I'm not convinced that adding this logic to `dash_ag_grid` or `dash_data_table` figure functions by default is the best approach. Instead, Vizro users can filter the data_frame before passing it to the data_manager, which could lead to even better performance improvements.

Reply:
Hm very good points! This is a tricky one, because I would guess that many people would provide a DF, then pick some columns with `columnDefs` and be done with it.

The problem with your suggested solution is that it does not cover a case that might also occur frequently: minimizing data loading from source, ie wanting to have one central DF loaded, but also customizing the table shown nonetheless.

Overall I think you are correct though, and we should not do what I proposed. Do you think we should work with docs and/or even some warning?

For the simple case I provided, maybe we could raise a warning that this may unnecessarily slow down the server response time or something similar?

Reply:
I've had an idea for a long time for working with big data in Vizro (and I'm already gathering and writing down good practices). I think that big-data handling in Vizro should get a whole page in the documentation, where this case will be one of the many items. 

What do you think? Do we have to emphasise anything just now, or we can wait until the "big-data handling in Vizro" doc page become added?

Reply:
If it is not too far away, then it's ok to now have it now haha. But it should come soon!

Reply:
Here's the [Issue](https://github.com/McK-Internal/vizro-internal/issues/1230) where I referenced this discussion and your commit along with some other recommendations we should add.